pkg/artifactory/resource_xray_watch.go

```diff
@@ -315,6 +321,11 @@ func resourceXrayWatchDelete(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).Xray
 
 	resp, err := c.V2.Watches.DeleteWatch(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.